src/imitation/algorithms/preference_comparisons.py

```diff
@@ -960,7 +969,7 @@ def push(self, fragments: Sequence[TrajectoryWithRewPair], preferences: np.ndarr
                 self.fragments2 = self.fragments2[extra:]
                 self.preferences = self.preferences[extra:]
 
-    def __getitem__(self, i) -> Tuple[TrajectoryWithRewPair, float]:
+    def __getitem__(self, i: int) -> Tuple[TrajectoryWithRewPair, float]:
```

Comment:
`i` can also be a slice object in general. Is there a reason why we're not allowing this here? (also, you would have to change the return type through an overload)

Reply:
Good idea, added!